src/main/java/org/opensearch/geospatial/ip2geo/common/DatasourceFacade.java

```diff
@@ -36,12 +41,37 @@
  */
 @Log4j2
 public class DatasourceFacade {
+    private static final Tuple<String, Integer> INDEX_SETTING_NUM_OF_SHARDS = new Tuple<>("index.number_of_shards", 1);
+    private static final Tuple<String, String> INDEX_SETTING_AUTO_EXPAND_REPLICAS = new Tuple<>("index.auto_expand_replicas", "0-all");
+    private static final Tuple<String, Boolean> INDEX_SETTING_HIDDEN = new Tuple<>("index.hidden", true);
     private final Client client;
+    private final ClusterService clusterService;
     private final ClusterSettings clusterSettings;
 
-    public DatasourceFacade(final Client client, final ClusterSettings clusterSettings) {
+    public DatasourceFacade(final Client client, final ClusterService clusterService) {
         this.client = client;
-        this.clusterSettings = clusterSettings;
+        this.clusterService = clusterService;
+        this.clusterSettings = clusterService.getClusterSettings();
+    }
+
+    /**
+     * Create a datasource index of single shard with auto expand replicas to all nodes
+     * We don't specify mapping explicitly here. It is controlled by {@code Datasource}
```

Comment:
If we put different mapping than what dynamic mapping put, it will break if we add a new field because old code will handle new field with dynamic mapping. Therefore, our mapping will be same as what dynamic mapping does. If that is the case, what would be the benefit of having mapping here when it will be same as dynamic mapping anyway.

Reply:
On a second thought, there could be some benefit of having explicit mapping for current field like disabling indexing for those field even if it won't apply for already existing index. Let me add a mapping.